storage/src/robocorp/storage/__init__.py

```diff
@@ -13,157 +16,157 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def _handle_missing_asset(response: requests.Response, *, name: str, asset_id: str):
+    assets_client = _get_assets_client()
+    try:
+        assets_client.handle_error(response)
+    except RequestsHTTPError as exc:
+        if exc.status_code == 404:
+            # NOTE(cmin764): Any `404 Not Found` error won't be retried and will
+            #  bubble-up from the first request.
+            message = f"Asset {name!r} with id {asset_id!r} not found"
+            raise AssetNotFound(
+                message, status_code=exc.status_code, status_message=exc.status_message
+            ) from exc
+        else:
+            raise exc
```

Comment:
We use this handler with bot GET and DELETE. (since the asset may not be found in both scenarios)